lis/surfacemodels/land/noahmp.4.0.1/irrigation/noahmp401_getirrigationstates.F90

```diff
@@ -175,341 +178,645 @@ subroutine noahmp401_getirrigationstates(n,irrigState)
      otimee = otimefs + irrhrf
   endif
   
-
-
-  do i=1,LIS_rc%npatch(n,LIS_rc%lsm_index)/LIS_rc%nensem(n)
-
-     sfctemp_avg = 0.
-     shdfac_avg  = 0.
-     smc_avg     = 0.
-
-     do m=1,LIS_rc%nensem(n)
-
-        t=(i-1)*LIS_rc%nensem(n)+m
-        
-        sfctemp_avg=sfctemp_avg+NOAHMP401_struc(n)%noahmp401(t)%sfctmp        
-        shdfac_avg=shdfac_avg+NOAHMP401_struc(n)%noahmp401(t)%fveg            
-        
-        do k=1,nsoil
-           
-           smc_avg(k)=smc_avg(k)+NOAHMP401_struc(n)%noahmp401(t)%smc(k)
-        
-        end do
+  if(LIS_rc%pert_bias_corr.eq.0) then
```

Comment:
I guess there needs to be an additional condition for soil moisture perturbations to be safe here. If LST, vegetation or snow or only forcings is/are perturbed over a pixel where irrigation can be applied, then we could theoretically still have meaningful ensemble irrigation estimates.

Reply:
True, will incorporate this.

Reply:
This is addressed now. Since the LIS da objects containing the information about perturbed state variable names are not available in the registered irrigation module, I decided to search "Soil Moisture" in the perturbation files which was less invasive than trying to change something more fundamental at the level of plugins and function registration. I tested this part and now an irrigation ensemble will always be generated when soil moisture is not perturbed.